### ISSUE TITLE
BuildFile.xml compilation issue with latest CMSSW versions

### ISSUE THREAD

**Issue by portalesHEP** [LLRCMS/LLRHiggsTauTau#132]
Compilation issue with `NtupleProducer/plugins/BuildFile.xml` and `NtupleProducer/BuildFile.xml`: the CMMSW version can't be parsed anymore from shell commands for some reason. A quick and dirty fix is to hardcode it, replacing
```
  <Flags CPPDEFINES="CMSSW_VERSION=$(shell echo ${CMSSW_VERSION}|sed -e s/CMSSW_// -e s/_//g -e s/patch\.//)"/>
```  
with 
```
<Flags CPPDEFINES="CMSSW_VERSION=$(10627)"/> 
```
but there should be a cleaner way to avoid the issue.

**Comment by portalesHEP** [comment]
@dzuolo @francescobrivio do you know if we actually need this? I tried commenting out the line, and everything seems to compile/run normally, but I'd like confirmation.

**Comment by francescobrivio** [comment]
@portalesHEP actually I did the same test last week and indeed everything compiles fine.
I have made two root file without this line and with your fix proposed above, but I still have to compare them, I'll try to do it today and let you know.

**Comment by portalesHEP** [comment]
Thanks @francescobrivio for providing the files. I did a quick check, and it'd seem that two branches (`neutralIsoPtSumWeight` and `photonPtSumOutsideSignalCone`) are different ***for one event***: 
```1
***********************************************************************************************************
*    Row   * Instance * EventNumb * footprint * t2.footpr * neutralIs * t2.neutra * photonPtS * t2.photon *
***********************************************************************************************************
*      313 *        0 * 174475811 * 1.8754992 * 1.8754992 * 0.8862836 * -2.20e+16 * 4.6072144 * 4.578e-41 * 
```
The `footprint` branch also does shows some differences which we can probably ignore ( some values set to 10^-41 instead of 0 )

**Comment by francescobrivio** [comment]
mmm that's not a good sign, they are completely different values for those quantities....

**Comment by portalesHEP** [comment]
I must admit my ignorance there: do you know where/how these variables are supposed to be used? I can see that they are retrieved from the bigNtuple in KLUB, but I can't find anywhere where they are used

**Comment by portalesHEP** [comment]
Copying investigations from Marcel here for reference:

> It looks like the value is added as a preprocessor variable during compilation to "$CMSSW_BASE/tmp/slc7_amd64_gcc820/src/LLRHiggsTauTau/NtupleProducer/src/LLRHiggsTauTauNtupleProducer/a/LLRHiggsTauTauNtupleProducer_xr.cc"
but it is never used anywhere. I found the identical "syntax" also elsewhere (e.g. https://github.com/cmstas/NtupleMaker/blob/combined/BuildFile.xml), so I guess some ppl found that useful in the past in order to make their code dependent on the CMSSW version and for that matter, checks with numeric values (resulting from that sed'ing) are easier than parsing the version string in C++ (I backtraced it to some egamma code where they do exactly that).
Also, Tobias found that the support for value injection was dropped: https://indico.cern.ch/event/1058840/contributions/4450329/attachments/2281171/3875942/CMSSDT-CoreSW-210713.pdf

**TLDR**: the shell reading functionality has been phased out, and there are good indications that setting the release flag serves no purpose.

Also, the three "problematic" variables, `footprintCorrection`,  `neutralIsoPtSumWeight` and `photonPtSumOutsideSignalCone` are saved for soft leptons, but not used anywhere in KLUB as far as I could tell. In addition, they are saved in the bigNtuples for any lepton flavors, but only filled for soft hadronic taus. The variables are initialized here: 

https://github.com/LLRCMS/LLRHiggsTauTau/blob/42b5a135dea738e0393f982d43b17f6be2466354/NtupleProducer/plugins/HTauTauNtuplizer.cc#L2876

without setting any default values, so in the cases where no value is attributed further down in the code (ie for soft muons/electrons), arbitrary floats are pushed into the saved vector.  This could be fixed easily, but since these variables have no apparent use (?), the better thing would probably be to remove them completely.

@dzuolo @francescobrivio please tell me if there is any good reason that I missed to keep the variables, otherwise I'll remove them.

**Comment by dzuolo** [comment]
Hi Louis! I am sure that those variables are never used in KLUB and have in the framework for such a long time that I am not even sure they exist in miniAOD. I vote for removing them.

**Comment by portalesHEP** [comment]
The variables are being removed in the PR https://github.com/LLRCMS/LLRHiggsTauTau/pull/137
